### PR TITLE
Incremental harvesting

### PR DESCRIPTION
**Pull request by jgonggrijp**
Implements #22.

### CODE REVIEW

File: oaiharvest/harvest.py [hunk 1]
```diff
@@ -207,11 +240,6 @@ def main(argv=None):
     else:
         args = argparser.parse_args(argv)
     logger = logging.getLogger(__name__).getChild('main')
-    # Parse from and until into datetime objects
```
Comment (bloomonkey) [inline oaiharvest/harvest.py:210]: Did you mean to remove the functionality to selectively harvest records updated between specified dates?

File: oaiharvest/harvest.py [hunk 2]
```diff
@@ -26,15 +26,23 @@
   -u YYYY-MM-DD, --until YYYY-MM-DD
                         harvest only records added/modified up to this date.
   -s SET, --set SET     harvest only records within this set
+  -b HH:MM HH:MM, --between HH:MM HH:MM
```
Comment (bloomonkey) [inline oaiharvest/harvest.py:31]: Does this actually work with shell command line parsing? Argument values aren't usually allowed to contain whitespace.

File: oaiharvest/harvest.py [hunk 3]
```diff
@@ -73,11 +82,33 @@ class OAIHarvester(object):
...
+        sleep((until - now) / timedelta(seconds=1))
+
+    def maybe_pause_if_incremental(self, time_range):
+        """ Pause the process depending on incremental time range settings. """
+        if time_range is None:
+            return
+        now = datetime.now()
+        start = datetime.combine(now.date(), time_range[0].time())
+        stop = datetime.combine(now.date(), time_range[1].time())
+        if now < start:
```
Comment (bloomonkey) [inline oaiharvest/harvest.py:100]: Please could you add some tests for the method to validate this logic?

**Review by bloomonkey — CHANGES_REQUESTED** [review]

File: oaiharvest/harvest.py (hunk 1 above)
Comment (jgonggrijp) [reply oaiharvest/harvest.py:210]: No, and I didn't. I moved the parsing over here:

https://github.com/bloomonkey/oai-harvest/blob/051668117f33822f4d6db5afe28b1a1325320afe/oaiharvest/harvest.py#L365-L367

https://github.com/bloomonkey/oai-harvest/blob/051668117f33822f4d6db5afe28b1a1325320afe/oaiharvest/harvest.py#L402-L405

https://github.com/bloomonkey/oai-harvest/blob/051668117f33822f4d6db5afe28b1a1325320afe/oaiharvest/harvest.py#L410-L413

File: oaiharvest/harvest.py (hunk 2 above)
Comment (jgonggrijp) [reply oaiharvest/harvest.py:31]: Yes, the `nargs=2` argument takes care of this.

https://github.com/bloomonkey/oai-harvest/blob/051668117f33822f4d6db5afe28b1a1325320afe/oaiharvest/harvest.py#L423-L427

File: oaiharvest/harvest.py (hunk 3 above)
Comment (jgonggrijp) [reply oaiharvest/harvest.py:100]: Sure. It may take a while before I get to it, though.

**Review by bloomonkey — APPROVED** [review]

**Pull request merged**